### DISCUSSION TITLE
Auto Update failed for v0.107.15

### DISCUSSION THREAD

**Discussion by sydsyd** [AdguardTeam/AdGuardHome#4989] [General]
On a Raspberry Pi - auto update is failing.
Haven't seen this issue before with other updates.
Just letting community know.

**Comment by tomhusband** [comment]
Yes, I can't update either. Installing manually is a bit intimidating for me. My Linux skills are pretty basic. Are there plans to correct this?

**Reply by sydsyd** [reply]
Updating manually on a Raspberry Pi didn't go well for me... I had to reinstall.

**Reply by tomhusband** [reply]
I really don't want to do that. I'll give it some time. Maybe there's a fix coming.

**Reply by threo9** [reply]
Waiting for fix too.

**Reply by derounrein** [reply]
Not only rasp. 3 installations - OL8, and two Ubuntu 22. Autoupdate fails everywhere. +1 for fix.

**Comment by ainar-g** [comment]
This is an unfortunate consequence of #4970.

Theoretically, something like this could work:

```sh
curl -H 'Content-Type: application/json' -X POST -v 'http://LOGIN:PASSWORD@127.0.0.1/control/update'
```

Where `LOGIN` and `PASSWORD` are your web interface login data and `127.0.0.1` is the address of AdGuard Home's web UI.  That successfully updates AdGuard Home from v0.107.14 to v0.107.15 for me.

**Reply by derounrein** [reply]
Wow, dats worked for me. Still needs a bugfix...

**Reply by ainar-g** [reply]
The mentioned bug has been fixed in v0.107.15.  This is merely a workaround for the bugged version.

**Reply by Jefffish09** [reply]
Thanks for solution, I met same problem.

**Reply by boyweiser** [reply]
thank you for the solution, i wasn't able to follow the re install guide !

**Reply by Flyinace2000** [reply]
When attempting this I get below error.  I have this installed on my OpnSense box.  I get the same error if I use 10.0.1.1 or 10.0.1.1:81.  Also tried with and without the http://.  Its like its trying to use the password as the login.  Could it be due to some special character?  I used a password manager to create the credential.

"Could Not Resolve Host: <long password>@10.0.1.1

**Reply by Flyinace2000** [reply]
Ended up being a "@" sign in my password that was causing the curl command to fail.  

Tried using the variable approach which got me closer.

MY_PASSWORD="@@@@@@@@@@@@@@@@@@@" curl -H 'Content-Type: application/json' -X POST -v 'http://Admin:${MY_PASSWORD}@10.0.1.1:81/control/update'

That resulted in the following results:
```
User-Agent: curl/7.79.1
> Accept: */*
> Content-Type: application/json
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 403 Forbidden
< Date: Thu, 06 Oct 2022 00:38:47 GMT
< Content-Length: 9
< Content-Type: text/plain; charset=utf-8
< 
* Connection #0 to host 10.0.1.1 left intact
Forbidden%                                     
```

**Reply by Robert-litts** [reply]
I was also having the same issue with the auto update failing. I tried the curl command and my password (containing special characters) was also failing. 

This post at Reddit helped solve my issue though: [Auto Update Fails](https://www.reddit.com/r/AdGuardHome/comments/xv3l4s/auto_update_fails/)

Specifically, I used [this encoder](https://www.urlencoder.org/ ) for my password, and then used the output as my password in the curl command. I also thought I would have to have Adguard stopped while performing this, but it is the opposite... Adguard must be running.

After doing this, I am now updated on v0.108.0-b.17.

**Reply by pyxis1** [reply]
Thanks for the help above! This got me a step further. But Still no go. It ends with:
/update request isn't allowed now
* Connection #0 to host 192.168.1.1 left intact
what am I missing?

**Solved
I was also logged in into adguard webpage while trying to use the command above to update. This did not work. While not being logged in and trying the command, it updated!

**Reply by Dimak47X** [reply]
```
*   Trying 192.168.1.205:80...
* Connected to 192.168.1.205 (192.168.1.205) port 80 (#0)
* Server auth using Basic with user 'admin'
> POST /control/update HTTP/1.1
> Host: 192.168.1.205
> Authorization: Basic YWRtaW46YWRtaW4x
> User-Agent: curl/7.79.1
> Accept: */*
> Content-Type: application/json
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 307 Temporary Redirect
< Location: https://192.168.1.205/control/update
< Date: Thu, 06 Oct 2022 09:35:03 GMT
< Content-Length: 0
< 
* Connection #0 to host 192.168.1.205 left intact
```
Is it ok or not? What should I do next? The version of Adguard is still the v0.107.14

**Reply by derounrein** [reply]
> ```
> *   Trying 192.168.1.205:80...
> * Connected to 192.168.1.205 (192.168.1.205) port 80 (#0)
> * Server auth using Basic with user 'admin'
> > POST /control/update HTTP/1.1
> > Host: 192.168.1.205
> > Authorization: Basic YWRtaW46YWRtaW4x
> > User-Agent: curl/7.79.1
> > Accept: */*
> > Content-Type: application/json
> > 
> * Mark bundle as not supporting multiuse
> < HTTP/1.1 307 Temporary Redirect
> < Location: https://192.168.1.205/control/update
> < Date: Thu, 06 Oct 2022 09:35:03 GMT
> < Content-Length: 0
> < 
> * Connection #0 to host 192.168.1.205 left intact
> ```
> 
> Is it ok or not? What should I do next? The version of Adguard is still the v0.107.14

It's OK when it prints "OK". Or when update done successfully :)

**Reply by Dimak47X** [reply]
> It's OK when it prints "OK". Or when update done successfully :)

Ok :) But I have applied the command above and changed only username, password and ip of my Raspberry Pi with AdGuard Home server. Maybe I should change something else in the command? Or maybe there are any other options to update the version without reinstalling AdGuard Home? Thanks.

**Reply by derounrein** [reply]
> > It's OK when it prints "OK". Or when update done successfully :)
> 
> Ok :) But I have applied the command above and changed only username, password and ip of my Raspberry Pi with AdGuard Home server. Maybe I should change something else in the command? Or maybe there are any other options to update the version without reinstalling AdGuard Home? Thanks.

Maybe you should use localhost or 127.0.0.1 instead of real local IP. Thats worked for me that way.

**Reply by Dimak47X** [reply]
> Maybe you should use localhost or 127.0.0.1 instead of real local IP. Thats worked for me that way.

If I use 127.0.0.1 then I get the error "connection refused"

*   Trying 127.0.0.1...
* TCP_NODELAY set
* connect to 127.0.0.1 port 80 failed: В соединении отказано
* Failed to connect to 127.0.0.1 port 80: В соединении отказано
* Closing connection 0
curl: (7) Failed to connect to 127.0.0.1 port 80: В соединении отказано

**Reply by Flyinace2000** [reply]
Should the curl command be run from another machine on the same network or while SSH'ed into the box running AdGuard?

**Reply by Dimak47X** [reply]
So I have found the problem. There was HTTPS encryption enabled on my AdGuard. In the web interface of AdGuard I went to Settings -> Encryption settings -> UNcheck "Enable encryption" and save configuration. After that the command above was completed successfully

> *   Trying 192.168.1.205:80...
> * Connected to 192.168.1.205 (192.168.1.205) port 80 (#0)
> * Server auth using Basic with user 'admin'
>  POST /control/update HTTP/1.1
>  Host: 192.168.1.205
>  Authorization: Basic YWRtaW46YWRtaW4=
>  User-Agent: curl/7.79.1
>  Accept: */*
>  Content-Type: application/json
>  
> * Mark bundle as not supporting multiuse
> < HTTP/1.1 200 OK
> < Access-Control-Allow-Origin: http://192.168.1.205
> < Vary: Origin
> < Vary: Accept-Encoding
> < Date: Thu, 06 Oct 2022 14:40:24 GMT
> < Content-Type: text/plain; charset=utf-8
> < Transfer-Encoding: chunked
> < 
> OK
> * Connection #0 to host 192.168.1.205 left intact

After that I was able to login to web console again (be sure to use http in the address bar this time, not https) and check that version is v0.107.15 now. HTTPS encryption may be enabled again then.

**Comment by masdeirf** [comment]
@Dimak47X: don't work for me, I don't have encryption enabled on my AdGuard. 😐

**Reply by Dimak47X** [reply]
Which error did you get when you run the command? May you copy-paste the output from the terminal?

**Comment by SmogUA** [comment]
![image](https://user-images.githubusercontent.com/17667693/194695946-63af46f8-82b6-4a00-9371-9b19c4fb7b2f.png)

**Reply by masdeirf** [reply]
```
Failed to load resource: the server responded with a status of 415 (Unsupported Media Type)
only application/json is allowed
```

**Comment by masdeirf** [comment]
> This is an unfortunate consequence of #4970.
> 
> Theoretically, something like this could work:
> 
> ```shell
> curl -H 'Content-Type: application/json' -X POST -v 'http://LOGIN:PASSWORD@127.0.0.1/control/update'
> ```
> 
> Where `LOGIN` and `PASSWORD` are your web interface login data and `127.0.0.1` is the address of AdGuard Home's web UI. That successfully updates AdGuard Home from v0.107.14 to v0.107.15 for me.

Worked for me 👍🙏 but I needed to add the port:

```shell
curl -H 'Content-Type: application/json' -X POST -v 'http://LOGIN:PASSWORD@127.0.0.1:PORT/control/update'
 ```

(HTTPS encryption was off)

**Reply by tomhusband** [reply]
@ainar-g Many thanks, this worked perfectly. I'm going to save the command. I might need it again someday.

**Reply by SmogUA** [reply]
Works fine for me. Thanks a lot!

**Comment by Flyinace2000** [comment]
So now that 0.107.16 I'm still stuck on dot14.  My password has an "@" contained within.  So if I run the command on my MacOS machine I tried the following command:

MY_PASSWORD=“12345@098765” curl -H 'Content-Type: application/json' -X POST -v 'http://Admin:${MY_PASSWORD}@10.0.1.1:81/control/update'


```
*   Trying 10.0.1.1:81...
* Connected to 10.0.1.1 (10.0.1.1) port 81 (#0)
* Server auth using Basic with user 'Admin'
> POST /control/update HTTP/1.1
> Host: 10.0.1.1:81
> Authorization: Basic 12345@098765=
> User-Agent: curl/7.79.1
> Accept: */*
> Content-Type: application/json
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 403 Forbidden
< Date: Mon, 10 Oct 2022 23:46:00 GMT
< Content-Length: 9
< Content-Type: text/plain; charset=utf-8
< 
* Connection #0 to host 10.0.1.1 left intact
Forbidden%
```

If I try a similar command while SSH into my OPNSense box which runs AdGuard. declaring the MY_PASSWORD variable doesn't work. Removing that and inserting the password back into the command returns the following error

```
root@OPNsense:~ # curl -H 'Content-Type: application/json' -X POST -v 'http://Admin:12345@098765@http://127.0.0.1:81/control/update'
* Closing connection -1
curl: (3) URL using bad/illegal format or missing URL

```

Https is not enabled.

**Reply by Flyinace2000** [reply]
Ended up just re-installing from the OPNSense plugin and reconfiguring my install.  Took the opportunity to make a new password to remove the "@".

**Reply by add0242** [reply]
Have the same problem, my password uses an ']'

**Comment by Gavrook** [comment]
Hi I have 2 instances of AdGuard Home installed on 2 Ubuntu Server machines both setup the same, they were both working fine for ages but one has for the past few months been giving me the titled error and won't update. The other is still working perfectly fine and receiving updates.

A while ago I managed to do a manual update, that got it to it's current version but the auto-update error still persisted.

I have tried all above and nothing works. Getting connection refused.
Encryption is off, but annoyingly my password does have an @ in it.
